src/Symfony/Component/HttpClient/Test/HarFileResponseFactory.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Test;
+
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Component\HttpClient\Response\MockResponse;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * See: https://w3c.github.io/web-performance/specs/HAR/Overview.html.
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class HarFileResponseFactory
+{
+    public function __construct(private string $archiveFile)
+    {
+    }
+
+    public function setArchiveFile(string $archiveFile): void
+    {
+        $this->archiveFile = $archiveFile;
+    }
+
+    public function __invoke(string $method, string $url, array $options): ResponseInterface
+    {
+        if (!is_file($this->archiveFile)) {
+            throw new \InvalidArgumentException(sprintf('Invalid file path provided: "%s".', $this->archiveFile));
+        }
+
+        $json = json_decode(
+            json: file_get_contents($this->archiveFile),
+            associative: true,
+            flags: \JSON_THROW_ON_ERROR
```

Comment:
```suggestion
            flags: \JSON_THROW_ON_ERROR,
```

Reply:
Let's be consistent with the code base and use the "short" version here on one line.

Reply:
Fixed!